src/ResultWriter/EnergyOutput.cpp

```diff
@@ -0,0 +1,142 @@
+  feKrnl.spaceWeights = spaceWeights;
+  feKrnl.frictionalEnergy = &staticWork;
+  feKrnl.timeWeight = -0.5 * godunovData.doubledSurfaceArea;
+  feKrnl.execute();
+
+  return staticWork;
+}
+
+void seissol::writer::printEnergies(GlobalData const* global,
+                                    seissol::initializers::DynamicRupture* dynRup,
+                                    seissol::initializers::LTSTree* dynRupTree,
+                                    MeshReader const& i_meshReader,
+                                    seissol::initializers::LTSTree* i_ltsTree,
+                                    seissol::initializers::LTS* i_lts,
+                                    seissol::initializers::Lut* i_ltsLut,
+                                    bool usePlasticity) {
+  double totalWorkLocal = 0.0;
+  double staticWorkLocal = 0.0;
+  double plasticMomentLocal = 0.0;
+  for (auto it = dynRupTree->beginLeaf(); it != dynRupTree->endLeaf(); ++it) {
+    /// \todo timeDerivativePlus and timeDerivativeMinus are missing the last timestep.
+    /// (We'd need to send the dofs over the network in order to fix this.)
+    real** timeDerivativePlus = it->var(dynRup->timeDerivativePlus);
+    real** timeDerivativeMinus = it->var(dynRup->timeDerivativeMinus);
+    DRGodunovData* godunovData = it->var(dynRup->godunovData);
+    DRFaceInformation* faceInformation = it->var(dynRup->faceInformation);
+    DROutput* drOutput = it->var(dynRup->drOutput);
+
+#ifdef _OPENMP
+#pragma omp parallel for reduction(+ : totalWorkLocal) reduction(+ : staticWorkLocal)
+#endif
+    for (unsigned i = 0; i < it->getNumberOfCells(); ++i) {
+      if (faceInformation[i].plusSideOnThisRank) {
+        totalWorkLocal += drOutput[i].frictionalEnergy;
+        staticWorkLocal += computeStaticWork(global,
+                                             timeDerivativePlus[i],
+                                             timeDerivativeMinus[i],
+                                             faceInformation[i],
+                                             godunovData[i],
+                                             drOutput[i].slip);
+      }
+    }
+  }
+  if (usePlasticity) {
+    plasticMomentLocal = computePlasticMoment(i_meshReader, i_ltsTree, i_lts, i_ltsLut);
+  }
+  int rank;
+  double totalWorkGlobal = 0.0;
+  double staticWorkGlobal = 0.0;
+  double plasticMomentGlobal = 0.0;
+#ifdef USE_MPI
+  MPI_Comm_rank(MPI::mpi.comm(), &rank);
+  MPI_Reduce(&totalWorkLocal, &totalWorkGlobal, 1, MPI_DOUBLE, MPI_SUM, 0, MPI::mpi.comm());
+  MPI_Reduce(&staticWorkLocal, &staticWorkGlobal, 1, MPI_DOUBLE, MPI_SUM, 0, MPI::mpi.comm());
+  MPI_Reduce(&plasticMomentLocal, &plasticMomentGlobal, 1, MPI_DOUBLE, MPI_SUM, 0, MPI::mpi.comm());
+#else
+  rank = 0;
+  totalWorkGlobal = totalWorkLocal;
+  staticWorkGlobal = staticWorkLocal;
+  plasticMomentGlobal = plasticMomentLocal;
+#endif
+
+  if (rank == 0) {
+    logInfo(rank) << "Total work:" << totalWorkGlobal;
```

Comment:
IMHO total work is misleading, as total work would be static work + frictional energy + plastic moment.

Reply:
ok changed to totalFrictionalWork